torchtext/datasets/text_classification.py

```diff
@@ -373,6 +446,39 @@ def AmazonReviewFull(*args, **kwargs):
     return _setup_datasets(*(("AmazonReviewFull",) + args), **kwargs)
 
 
+def IMDB(*args, **kwargs):
+    """ Defines IMDB datasets.
+        The labels includes:
+            - 0 : Negative
+            - 1 : Positive
+
+    Create sentiment analysis dataset: IMDB
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        root: Directory where the datasets are saved. Default: ".data"
+        ngrams: a contiguous sequence of n items from s string text.
+            Default: 1
+        vocab: Vocabulary used for dataset. If None, it will generate a new
+            vocabulary based on the train data set.
+        removed_tokens: removed tokens from output dataset (Default: [])
+        tokenizer: the tokenizer used to preprocess raw text data.
+            The default one is basic_english tokenizer in fastText. spacy tokenizer
+            is supported as well. A custom tokenizer is callable
+            function with input of a string and output of a token list.
+
+    Examples:
+        >>> from torchtext.datasets import IMDB
+        >>> from torchtext.data.utils import get_tokenizer
+        >>> train_dataset, test_dataset = IMDB(ngrams=3)
+        >>> tokenizer = get_tokenizer("spacy")
+        >>> train_dataset, test_dataset = IMDB(tokenizer=tokenizer)
+    """
+
+    return _setup_datasets(*(('IMDB',) + args), **kwargs)
```

Comment:
Would `_setup_datasets('IMDB', *args, **kwargs)` do the same?

Reply:
yes. We decide to let all the dataset call the same function `_setup_datasets` but with a different function `_imdb_iterator` to read data from file.

Reply:
@vincentqb - as far as I remember that syntax isn't supported by Python2.

Reply:
Glad we deprecate python 2 soon :)